fuel-crypto/src/tests/signature.rs

```diff
@@ -5,11 +5,14 @@ use crate::{
     SecretKey,
     Signature,
 };
+
+#[cfg(feature = "std")]
 use rand::{
     rngs::StdRng,
     SeedableRng,
 };
 
+#[cfg(feature = "std")]
```

Comment:
I get that the secret key generation doesn't work in no_std, but do we have tests in other places to ensure that the no_std backends work the same as std?

Reply:
The test for backend equivalency is in `fuel-crypto/src/secp256/backend.rs`. Only the secp256k1 has different implementations currently.

I agree that it makes sense to add some tests to the backends themselves.

Reply:
I added standalone test cases for k256 module that also run in `no_std` mode, in e3943dee